cli/src/cmd/forge/script/executor.rs

```diff
@@ -182,8 +182,15 @@ impl ScriptArgs {
                         runner.executor.env_mut().block.number += U256::one();
                     }
 
-                    // We inflate the gas used by the user specified percentage
-                    tx.gas = Some(U256::from(result.gas_used * self.gas_estimate_multiplier / 100));
+                    let is_fixed_gas_limit = tx.gas.is_some();
```

Comment:
At this point `tx.gas` is what was set by the inspector code [here](https://github.com/ckoopmann/foundry/blob/12d698ebe28ca467a9a93c0cf4f48ecc04b56f74/evm/src/executor/inspector/cheatcodes/mod.rs#L450).  
So if the gas_limit was  set manually then it will be that value otherwise it will be `none` (which is always the case in the current implementation on master).